packages/router/useRouteQuery/index.ts

```diff
@@ -34,18 +34,27 @@ export function useRouteQuery<
     transform = value => value as any as K,
   } = options
 
-  return computed<any>({
+  _query[name] = route.query[name]
+
+  tryOnScopeDispose(() => {
+    _query = {}
```

Comment:
Can you also test what would happend if you use mutliple `useRouteQuery` while some of them get disposed earlier (say they are used in different components that mounted/destoried differently)